datafusion/core/src/dataframe/mod.rs

```diff
@@ -896,9 +896,8 @@ impl DataFrame {
         join_type: JoinType,
         on_exprs: impl IntoIterator<Item = Expr>,
     ) -> Result<DataFrame> {
-        let expr = on_exprs.into_iter().reduce(Expr::and);
```

Comment:
It's redundant because the reduce operation will be performed inside the [join_on](https://github.com/apache/datafusion/blob/1dfac86a89750193491cf3e04917e37b92c64ffa/datafusion/expr/src/logical_plan/builder.rs#L677) function.